src/Features/Core/Portable/InlineParameterNameHints/IInlineParameterNameHintsService.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.InlineParameterNameHints
+{
+    internal readonly struct InlineParameterHint
+    {
+        public InlineParameterHint(string name, int position)
+        {
+            Name = name;
+            Position = position;
+        }
+        public string Name { get; }
+        public int Position { get; }
+    }
+
+    internal interface IInlineParameterNameHintsService : ILanguageService
+    {
+        Task<IEnumerable<InlineParameterHint>> GetInlineParameterNameHintsAsync(Document document, TextSpan textSpan, CancellationToken cancellationToken = default);
```

Comment:
```suggestion
        Task<IEnumerable<InlineParameterHint>> GetInlineParameterNameHintsAsync(Document document, TextSpan textSpan, CancellationToken cancellationToken);
```

Reply:
For internal APIs, prefer non-optional cancellation tokens.  We would prefer to not accidentally forget to pass this in.